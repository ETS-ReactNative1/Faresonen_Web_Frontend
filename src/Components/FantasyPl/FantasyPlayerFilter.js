import React, { Component } from "react";
import PropTypes from "prop-types";
import { Segment, Form, Checkbox } from "semantic-ui-react";
import "./FantasyPlayerFilter.css";

class FantasyPlayerFilter extends Component {
  handleNameSearch = search =>
    this.props.setNameFilter(search.target.value.trim().toLowerCase());

  handlePriceSearch = search => {
    const { value } = search.target;
    if (isNaN(value) || !value) this.props.setPriceFilter(-1);
    else {
      this.props.setPriceFilter(parseFloat(value, 10));
    }
  };

  handleDreamTeamFilter = (event, data) => {
    this.props.setDreamTeamFilter(data.checked);
  };

  render() {
    return (
      <Segment>
        <p className="special">
          Her kan du filtrere spillere ved å søke etter fornavn/etternavn eller
          filtrere på pris.
        </p>
        <Form>
          <Form.Field>
            <label>Søk fritekst</label>
            <input
              placeholder="Søk etter navn"
              onChange={this.handleNameSearch}
            />
          </Form.Field>
          <Form.Field>
            <label>Filtrer på pris i pund (£)</label>
            <input type="number" step={0.1} onChange={this.handlePriceSearch} />
          </Form.Field>
          <Form.Field>
            <Checkbox
              label="Vis kun Dream Team"
              onChange={this.handleDreamTeamFilter}
            />
          </Form.Field>
        </Form>
      </Segment>
    );
  }
}

FantasyPlayerFilter.propTypes = {
  setPriceFilter: PropTypes.func.isRequired,
  setNameFilter: PropTypes.func.isRequired,
  setDreamTeamFilter: PropTypes.func.isRequired
};

export default FantasyPlayerFilter;
