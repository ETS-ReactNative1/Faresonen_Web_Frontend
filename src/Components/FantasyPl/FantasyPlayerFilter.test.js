import React from "react";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";
import FantasyPlayerFilter from "./FantasyPlayerFilter";

describe("<FantasyPlayerFilter />", () => {
  it("Should render properly", () => {
    const tree = renderer.create(
      <FantasyPlayerFilter
        setNameFilter={() => {}}
        setPriceFilter={() => {}}
        setDreamTeamFilter={() => {}}
      />,
    );
    expect(tree).toMatchSnapshot();
  });

  it("Should handle the name search", () => {
    const setNameFilterMock = jest.fn();

    const value = {target: {value: "Testing"}};
    const elem = shallow(
      <FantasyPlayerFilter
        setPriceFilter={() => {}}
        setNameFilter={setNameFilterMock}
        setDreamTeamFilter={() => {}}
      />,
    );
    elem.instance().handleNameSearch(value);
    expect(setNameFilterMock).toBeCalled();
  });
  it("Should handle the price search", () => {
    const setPriceFilter = jest.fn();

    const value = {target: {value: 5}};
    const elem = shallow(
      <FantasyPlayerFilter
        setPriceFilter={setPriceFilter}
        setNameFilter={() => {}}
        setDreamTeamFilter={() => {}}
      />,
    );
    elem.instance().handlePriceSearch(value);
    expect(setPriceFilter).toBeCalled();

    elem.instance().handlePriceSearch({target: {value: "hey"}});
    expect(setPriceFilter).toBeCalledWith(-1);
  });

  it("Should handle the dream team filter", () => {
    const setDreamTeamFilterMock = jest.fn();

    const elem = shallow(
      <FantasyPlayerFilter
        setPriceFilter={() => {}}
        setNameFilter={() => {}}
        setDreamTeamFilter={setDreamTeamFilterMock}
      />,
    );
    elem.instance().handleDreamTeamFilter(null, {checked: true});
    expect(setDreamTeamFilterMock).toBeCalled();
  });
});
