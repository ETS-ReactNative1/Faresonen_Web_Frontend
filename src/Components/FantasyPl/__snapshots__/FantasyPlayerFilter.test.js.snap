// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FantasyPlayerFilter /> Should render properly 1`] = `
<div
  className="ui segment"
>
  <p
    className="special"
  >
    Her kan du filtrere spillere ved å søke etter fornavn/etternavn eller filtrere på pris.
  </p>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        Søk fritekst
      </label>
      <input
        onChange={[Function]}
        placeholder="Søk etter navn"
      />
    </div>
    <div
      className="field"
    >
      <label>
        Filtrer på pris i pund (£)
      </label>
      <input
        onChange={[Function]}
        step={0.1}
        type="number"
      />
    </div>
    <div
      className="field"
    >
      <div
        className="ui checkbox"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
      >
        <input
          checked={false}
          className="hidden"
          id={undefined}
          name={undefined}
          onClick={[Function]}
          readOnly={true}
          tabIndex={0}
          type="checkbox"
          value={undefined}
        />
        <label
          htmlFor={undefined}
        >
          Vis kun Dream Team
        </label>
      </div>
    </div>
  </form>
</div>
`;
